app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/model/quote-ext.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'ko',
+    'Magento_InventoryInStorePickupFrontend/js/model/pickup-address-converter',
+], function(ko, pickupAddressConverter) {
+    'use strict';
+
+    return function(quote) {
+        var shippingAddress = quote.shippingAddress;
+        /**
+         * Makes sure that shipping address gets appropriate type when it points
+         * to a store pickup location.
+         */
+        quote.shippingAddress = ko.pureComputed({
+            read: function() {
+                return shippingAddress();
+            },
+            write: function(address) {
```

Comment:
From my understanding, we will always convert address in the quote to pickupAddress format. Are you sure that this will not break anything in the default checkout flow?